ibis_bigquery/backports.py

```diff
@@ -0,0 +1,132 @@
+"""Backports for changes in Ibis."""
+
+import ibis
+import packaging
+from ibis.backends.base.sql.compiler import query_builder
+from ibis.backends.base.sql.compiler.base import QueryAST, SetOp
+from ibis.expr import operations as ops
+
+IBIS_VERSION = packaging.version.Version(ibis.__version__)
+IBIS_3_0_2_VERSION = packaging.version.Version("3.0.2")
+
+
+def import_default(module_name, force=False, default=None):
+    """
+    Provide an implementation for a class or function when it can't be imported
+    or when force is True.
+
+    This is used to replicate Ibis APIs that are missing or insufficient
+    (thus the force option) in early Ibis versions.
+
+    Pulled from
+    https://github.com/googleapis/python-db-dtypes-pandas/blob/1d32339fc72666d671d8d2dbca32e39abb3de94f/db_dtypes/pandas_backports.py#L50:5
+    """
+
+    if default is None:
+        return lambda func_or_class: import_default(module_name, force, func_or_class)
+
+    if force:
+        return default
+
+    name = default.__name__
+    try:
+        module = __import__(module_name, {}, {}, [name])
+    except ModuleNotFoundError:
+        return default
+
+    return getattr(module, name, default)
+
+
+@import_default("ibis.backends.base.sql.compiler", IBIS_VERSION <= IBIS_3_0_2_VERSION)
+class Difference(SetOp):
+    _keyword = "EXCEPT"
+
+    def _get_keyword_list(self):
+        return [self._keyword] * (len(self.tables) - 1)
+
+
+@import_default("ibis.backends.base.sql.compiler", IBIS_VERSION <= IBIS_3_0_2_VERSION)
+class Intersection(SetOp):
+    _keyword = "INTERSECT"
+
+    def _get_keyword_list(self):
+        return [self._keyword] * (len(self.tables) - 1)
+
+
+# Override Compiler to use overrides for Difference and Intersections. See
+# https://github.com/ibis-project/ibis-bigquery/issues/87 and
+# https://github.com/ibis-project/ibis/issues/3863
+if IBIS_VERSION <= IBIS_3_0_2_VERSION:
```

Comment:
Note to self, keep this in sync if there are any changes to https://github.com/ibis-project/ibis/pull/4006